lightning/src/ln/functional_test_utils.rs

```diff
@@ -1871,10 +1973,24 @@ pub fn do_claim_payment_along_route<'a, 'b, 'c>(origin_node: &Node<'a, 'b, 'c>,
 		}
 	}
 	let mut per_path_msgs: Vec<((msgs::UpdateFulfillHTLC, msgs::CommitmentSigned), PublicKey)> = Vec::with_capacity(expected_paths.len());
-	let events = expected_paths[0].last().unwrap().node.get_and_clear_pending_msg_events();
+	let mut events = expected_paths[0].last().unwrap().node.get_and_clear_pending_msg_events();
 	assert_eq!(events.len(), expected_paths.len());
-	for ev in events.iter() {
-		per_path_msgs.push(msgs_from_ev!(ev));
+
+	let mut expected_paths_msg_order = vec![origin_node.node.get_our_node_id()];
+
+	for expected_path in expected_paths.iter() {
+		for path_node in expected_path.iter() {
+			expected_paths_msg_order.push(path_node.node.get_our_node_id());
+		}
+	}
+
+	for msg_node_id in expected_paths_msg_order.iter() {
+		// Note that the expected_paths_msg_order contains more nodes than msg_events in events,
+		// as it's only the intended order for the messages that actually exists.
+		if let Some((ev, updated_events)) = try_remove_first_msg_event_to_node(*msg_node_id, &events) {
+			per_path_msgs.push(msgs_from_ev!(&ev));
+			events = updated_events;
+		}
```

Comment:
There's probably a much better and efficient way retrieve the messages from the `events` in the order we want here, but I couldn't figure out another way that works for all test scenarios that `do_claim_payment_along_route` is called under. Let me know if you can think of a better way, or if this is so dirty that I need to fix it.